plugins/carbon_black_defense/komand_carbon_black_defense/actions/find_event/action.py

```diff
@@ -1,68 +1,79 @@
-                Output.LATESTTIME: data["latestTime"],
-                Output.RESULTS: data["results"],
-                Output.ELAPSED: data["elapsed"],
-                Output.MESSAGE: data["message"],
-                Output.TOTALRESULTS: data["totalResults"],
-            }
-        if result.status_code in range(400, 499):
-            raise Exception(
-                f"Carbon Black returned a {result.status_code} code."
-                f" Verify the token and host configuration in the connection. Response was: {result.text}"
-            )
-        if result.status_code in range(500, 599):
-            raise Exception(
-                f"Carbon Black returned a {result.status_code} code."
-                f" If the problem persists please contact support for help. Response was: {result.text}"
+        criteria = {}
+        exclusions = {}
+
+        if device_external_ip:
+            criteria["device_external_ip"] = device_external_ip
+        if process_name:
+            criteria["process_name"] = [process_name]
+        if enriched_event_type:
+            criteria["enriched_event_type"] = enriched_event_type
+        if process_hash:
+            criteria["process_hash"] = [process_hash]
+        if device_name:
+            criteria["device_name"] = [device_name]
+        if len(criteria) == 0:
+            raise PluginException(
+                cause="Error. Have not entered a criteria for action to run.",
+                assistance="Enter a criteria of at least one plugin input.",
             )
-        self.logger.error(result.text)
-        raise Exception(
-            f"An unknown error occurred."
-            f" Carbon Black returned a {result.status_code} code. Contact support for help. Raw response in logs."
-        )
+        elif len(exclusions) > 0:
+            if device_external_ip:
+                exclusions["device_external_ip"] = device_external_ip
+            if process_name:
+                exclusions["process_name"] = [process_name]
+            if enriched_event_type:
+                exclusions["enriched_event_type"] = enriched_event_type
+            if process_hash:
+                exclusions["process_hash"] = [process_hash]
+            if device_name:
+                exclusions["device_name"] = [device_name]
+
+
+        id_ = self.connection.get_job_id_for_enriched_event(criteria, exclusions=exclusions)
+
+        self.logger.info(f"Got enriched event job ID: {id_}")
+        if id_ is None:
+            return {Output.EVENTINFO: {}}
+        enriched_event_search_status = self.connection.get_enriched_event_status(id_)
+
+        t1 = datetime.now()
+        for _ in range(0, 9999):
+            if not enriched_event_search_status:
+                enriched_event_search_status = self.connection.get_enriched_event_status(id_)
+                if (datetime.now() - t1).seconds > 60:
+                    break
+            else:
+                break
```

Comment:
From a scalability point of view, I think sleeping the thread here would be good. Open to other peoples suggestions though.